ci-operator/step-registry/ibmcloud/provision/dedicatedhost/ibmcloud-provision-dedicatedhost-commands.sh

```diff
@@ -0,0 +1,205 @@
+    dhg="${dhName}g"
+    dhg=$(${IBMCLOUD_CLI} is dedicated-host-group-create --zone $zone --family $family --class $class --name ${dhg} --output JSON | jq -r .id)
+    
+    ${IBMCLOUD_CLI} is dhg $dhg
+
+    ${IBMCLOUD_CLI} is dedicated-host-create --profile $dProfile --name $dhName --dhg $dhg --output JSON
+    
+    status=$(waitingStatus ${dhName})
+    echo "${dhName} status: ${status}"
+    run_command "${IBMCLOUD_CLI} is dedicated-host ${dhName}"
+
+    if [[ "${status}" != "stable" ]]; then
+        echo "ERROR: fail to create the dedicated host with profile $dProfile on $zone $rg"
+        deleteDedicatedHost "${dhName}" "${dhg}"
+    else
+        echo "${dhName}" > "${SHARED_DIR}/dedicated_host"
+        echo "${dhg}" > "${SHARED_DIR}/dedicated_host_group"
+    fi
+}
+
+function waitingStatus() {
+    local dh=$1 status
+    COUNTER=0
+    while [ $COUNTER -lt 200 ]
+    do 
+        sleep 10
+        COUNTER=`expr $COUNTER + 20`
+        status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+        if [[ "${status}" != "pending" ]]; then
+            break
+        fi
+    done
+    echo ${status}
+}
+
+function deleteDedicatedHost() {
+    local dh=$1 dhg=$2 status
+    status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+
+    run_command "${IBMCLOUD_CLI} is dhd ${dh} -f"
+    run_command "${IBMCLOUD_CLI} is dhgd ${dhg} -f"
+}
+
+function getVMProfile() {
+    local profileHeader=$1
+    local all_profiles str1 str2
+    all_profiles=$(${IBMCLOUD_CLI} is instance-profiles -q | awk '(NR>1) {print $1}')
+    str1=${all_profiles##*$profileHeader}
+    str2=$(echo ${str1}|cut -d " " -f1)
+    echo ${profileHeader}${str2}
+}
+
+#####################################
+##############Initialize#############
+#####################################
+ibmcloud_login
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+resource_group="${CLUSTER_NAME}-rg"
+create_resource_group ${resource_group}
+echo "${resource_group}" > "${SHARED_DIR}/ibmcloud_resource_group"
+
+#####################################
+#######create the dedicated host#####
+#####################################
+
+## Create the dedicated host
+echo "$(date -u --rfc-3339=seconds) - Creating the Dedicated Host..."
+
+dhName="${CLUSTER_NAME}-dh"
+dhProfile1=$(randomGetDHProfile)
+echo "random selected DHProfile1: $dhProfile1"
+
+dhGroup=$(randomSelectRG)
+echo "random select ResourceGroup of Dedicated Host [${dhGroup}]"
+
+zones=("${region}-1" "${region}-2" "${region}-3")
+for zone1 in "${zones[@]}"; do
+    createDH ${dhGroup} ${dhProfile1} ${zone1} ${dhName}
+    if [ -f "${SHARED_DIR}/dedicated_host" ]; then
+        zone=${zone1}
+        break
+    fi
+done
+
+class1=${dhProfile1%%-*}
+
+dhProfile2=$(randomGetDHProfile)
+echo "random selected DHProfile2: $dhProfile2"
+class2=${dhProfile2%%-*}
+
+#Know Issue OCPBUGS-5906 [IPI-IBMCloud] the worker nodes are “Provisioning” when using the dedicated host in the Default group
```

Comment:
So once random selected default resource group for DH, the install would failed?

Reply:
if the master nodes on the dh, which dhg is in the Default group, and worker nodes on the dh, which dhg is the cluster group, install cluster succeeds. 
ref https://mastern-jenkins-csb-openshift-qe.apps.ocp-c1.prod.psi.redhat.com/job/ocp-common/job/Flexy-install/233120/parameters/